http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -572,6 +572,23 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
       )(onError)
     } yield resp.map(_.objectId)
 
+  override def getResourceParent(authHeader: Authorization, resource: SamResourceId)(implicit
```

Comment:
Would it be possible to use the Sam client instead? We are trying to actively move away from using our custom DAO layer

Reply:
Yeah, long term I really want to scrap our existing auth layer (`HttpSamDAO`, `SamAuthProvider`, `LeoAuthProvider`, `samModels.scala`, etc) in favor of a simplified `SamService` class that uses the generated Sam client. 

I tried to fit this PR into the current pattern, but instead I could try to pull in the Sam client and use that instead. Then we could expand use of the client over time. I can give it a try..

Reply:
Added skaffolding for using the Sam client! Ported over some functionality (not any access control logic yet) as a POC.